src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,367 @@
+    ) ECDSAStakeRegistryStorage(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    function initialize(
+        address _serviceManager,
+        uint256 _thresholdWeightBps,
+        Quorum memory _quorum
+    ) external initializer {
+        __ECDSAStakeRegistry_init(_serviceManager, _thresholdWeightBps, _quorum);
+    }
+
+    /// @notice Registers a new operator using a provided signature
+    /// @param _operatorSignature Contains the operator's signature, salt, and expiry
+    function registerOperatorWithSignature(
+        address _operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory _operatorSignature
+    ) external {
+        _registerOperatorWithSig(_operator, _operatorSignature);
+    }
+
+    /// @notice Deregisters an existing operator
+    function deregisterOperator() external {
+        _deregisterOperator(msg.sender);
+    }
+
+    /**
+     * @notice Updates the StakeRegistry's view of one or more operators' stakes.
+     * @dev Queries stakes from the Eigenlayer core DelegationManager contract
+     * @param _operators A list of operator addresses to update
+     */
+    function updateOperators(address[] memory _operators) external {
+        for (uint256 i; i < _operators.length; i++) {
+            _updateOperatorWeight(_operators[i]);
+        }
+    }
+
+    /// @notice Updates the strategies and their weights for the quourum
+    /// @dev Access controlled to the contract owner
+    /// @param _quorum The new quorum configuration
+    function updateQuorumConfig(Quorum memory _quorum) external onlyOwner {
+        _updateQuorumConfig(_quorum);
+    }
+
+    /// @notice Updates the weight an operator must have to join the operator set
+    /// @dev Access controlled to the contract owner
+    /// @param _newMinimumWeight The new weight an operator must have to join the operator set
+    function updateMinimumWeight(uint256 _newMinimumWeight) external onlyOwner {
+        minimumWeight = _newMinimumWeight;
+    }
+
+    /**
+     * @notice Adjusts the threshold weight for valid signatures.
+     * @param _thresholdWeightBps The updated threshold weight.
+     */
+    function updateStakeThreshold(uint256 _thresholdWeightBps) external onlyOwner {
+        _updateStakeThreshold(_thresholdWeightBps);
+    }
+
+    function isValidSignature(
+        bytes32 _dataHash,
+        bytes memory _signatureData
+    ) external view returns (bytes4) {
+        (address[] memory signers, bytes[] memory signatures, uint32 referenceBlock) = abi.decode(
+            _signatureData,
+            (address[], bytes[], uint32)
+        );
+        _checkSignatures(_dataHash, signers, signatures, referenceBlock);
+        return IERC1271Upgradeable.isValidSignature.selector;
+    }
+
+    /// @notice Retrieves the current stake quorum details.
+    /// @return Quorum - The current quorum of strategies and weights
+    function stakeQuorum() external view returns (Quorum memory) {
```

Comment:
why is this called stakeQuorum as opposed to just quorum? Might make users wonder if this is different from the other quorums we are using elsewhere?

Reply:
Will need to update the internal variable. It conflicted so I just named it stakeQuorum